aioxmpp/avatar/service.py

```diff
@@ -337,35 +338,37 @@ def __init__(self, client, **kwargs):
         self._pubsub = self.dependencies[pubsub.PubSubClient]
 
     def _cook_metadata(self, jid, metadata):
+        def iter_metadata_info_nodes(metdata):
+            for item in metadata.payload.items:
+                info_map = item.registered_payload.info
+                for mime_type in info_map:
+                    for metadata_info_node in info_map[mime_type]:
+                        yield metadata_info_node
```

Comment:
The code looks much better this way!

Suggestion: You could refactor the last three lines into a property/method of the avatar metadata XSO.